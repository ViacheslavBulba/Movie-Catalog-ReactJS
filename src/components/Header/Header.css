.header-container {
    width: 100vw;
    display: flex;
    flex-direction: row;
    height: 100px;
    justify-content: space-between;
    align-items: center;
}

.header-container::before {
    content: '';
    position: absolute;
    left: 0;
    right: 0;
    z-index: -1;

    display: block;
    background-image: url('../../../public/netflix-background.jpg');
    background-size: cover;
    width: 100vw;
    height: 100%;

    -webkit-filter: blur(3px);
    -moz-filter: blur(3px);
    -o-filter: blur(3px);
    -ms-filter: blur(3px);
    filter: blur(3px) brightness(0.5);
}

.logo-container {
    margin-left: 20px;
}

.search-input {
    width: 50%;
    min-width: 250px;
    height: 40px;
    border-radius: 3px;
    border: solid 1px #d9dbe3;
    background-color: #ffffff;
    padding-left: 10px;
    font-family: 'Source Sans Pro';
    font-size: 18px;
}

.search-button {
    width: 150px;
    height: 40px;
    border-radius: 5px;
    font-family: 'Source Sans Pro';
    font-size: 18px;
    color: #ffffff;
    background-color: #f65261;
}

.reset-search-button {
    width: 150px;
    height: 40px;
    border-radius: 5px;
    font-family: 'Source Sans Pro';
    font-size: 18px;
    background-color: #ffffff;
    color: #f65261;
}

.add-button {
    padding: 0;
    width: 150px;
    height: 40px;
    border-radius: 5px;
    font-family: 'Source Sans Pro';
    font-size: 18px;
    color: #f65261;
    background-color: #555555;
    margin-right: 30px;
}