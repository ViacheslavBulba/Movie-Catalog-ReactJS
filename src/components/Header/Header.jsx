import React, { useState } from 'react';
import './Header.css';
import logo from '../../../public/netflix-logo.svg';
import MovieDetailsModal from '../shared/MovieDetailsModal/MovieDetailsModal';
import { setSearch, setSearchBy, thunkedSetMovies } from '../../store/actions';
import store from '../../store/store';

export default function Header() {
    const [showModal, setShowModal] = useState(false);

    const [searchValue, setSearchValue] = useState('');

    const handleCloseModal = () => {
        setShowModal(false);
    };

    const showAddMovieModal = () => setShowModal(true);

    const handleSearch = () => {
        store.dispatch(setSearch(searchValue));
        store.dispatch(setSearchBy('title')); // re-setting it to 'title' here in case it is changed by other component
        store.dispatch(thunkedSetMovies());
    };

    const doSearchIfEnterKeyPressed = (e) => {
        if (e.which === 13) {
            handleSearch();
        }
    };

    return (
        <header className='header-container'>
            <div className='logo-and-add-button-container'>
                <img src={logo} alt='logo' className='logo' />
                <button className='add-button' onClick={showAddMovieModal}>
                    ADD MOVIE
                </button>
                <MovieDetailsModal
                    show={showModal}
                    handleCloseModal={handleCloseModal}
                />
            </div>
            <div className='header-text'>FIND YOUR MOVIE</div>
            <div className='search-container'>
                <input
                    type='text'
                    placeholder='What do you want to watch?'
                    className='search-input'
                    value={searchValue}
                    onChange={(e) => setSearchValue(e.target.value)}
                    onKeyPress={doSearchIfEnterKeyPressed}
                />
                <button className='search-button' onClick={handleSearch}>
                    SEARCH
                </button>
            </div>
        </header>
    );
}
