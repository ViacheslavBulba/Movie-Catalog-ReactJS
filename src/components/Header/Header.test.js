import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

const initialState = {
    pending: false,
    movies: [],
    error: null,
    sortBy: 'release_date',
    sortOrder: 'desc',
    filerByGenres: [],
    search: '',
    searchBy: 'title',
    movieToOverview: null,
};

const store = mockStore(initialState)

test('snapshot test for header', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Header />
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
