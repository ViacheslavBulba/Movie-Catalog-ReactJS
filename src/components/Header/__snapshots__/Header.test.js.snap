// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for header 1`] = `
<header
  className="header-container"
>
  <div
    className="logo-and-add-button-container"
  >
    <img
      alt="logo"
      className="logo"
      src="netflix-logo.svg"
    />
    <button
      className="add-button"
      onClick={[Function]}
    >
      ADD MOVIE
    </button>
  </div>
  <div
    className="header-text"
  >
    FIND YOUR MOVIE
  </div>
  <div
    className="search-container"
  >
    <input
      className="search-input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <button
      className="search-button"
      onClick={[Function]}
    >
      SEARCH
    </button>
  </div>
</header>
`;
